sql/table_expression_operations.cc

```diff
@@ -645,15 +646,53 @@ bind(SqlBindingScope & context) const
 
     auto boundExpr = expr->bind(context);
 
-    // TODO: infer value expression from row, especially if there is only
+    // infer value expression from row, especially if there is only
     // one column available and thus its value type is simple
+    auto inputInfo = ExpressionValueInfo::toRow(boundExpr.info);
 
+    std::shared_ptr<ExpressionValueInfo> valueInfo;
+    
+    switch (style) {
+    case COLUMNS:
+        if (inputInfo->getSchemaCompleteness() == SCHEMA_CLOSED) {
+            // Try to infer a type for the column column
+            auto knownColumns = inputInfo->getKnownColumns();
+            if (!knownColumns.empty()) {
+                valueInfo = knownColumns[0].valueInfo;
+                for (size_t i = 1;  i < knownColumns.size();  ++i) {
+                    valueInfo = ExpressionValueInfo
```

Comment:
Unsure (I don't know what getCovering does.) Is it intended to loop to assign to `valueInfo` every time? Couldn't it just take the last value instead of looping?